src/pages/index.js

```diff
@@ -59,54 +32,126 @@ const config = {
+const profileImage = document.querySelector(".profile__image");
+const profileAvatarModalForm = profileAvatarModal.querySelector(".modal__form");
+
+const userInfo = new UserInfo(".profile__title", ".profile__description");
+const cardSection = new Section(renderCard, cardListEl);
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "f1dab6ed-e5ba-44ef-bef1-7369d7e9bf0d",
+    "Content-Type": "application/json",
+  },
+});
+
+api.getUserInfo().then((res) => {
+  userInfo.setUserInfo(res);
+});
+api.getInitialCards().then((res) => {
+  cardSection.renderItems(res);
+});
+api.getUserInfo().then((res) => {
+  profileImage.src = res.avatar;
+});
+
+const cardDeleteModal = new PopupQuestion("#delete-card-modal");
 
 const editFormValidator = new FormValidator(config, profileEditForm);
 editFormValidator.enableValidation();
 const addFormValidator = new FormValidator(config, addCardFormElement);
 addFormValidator.enableValidation();
+const profileAvatarFormValidator = new FormValidator(
+  config,
+  profileAvatarModalForm
+);
 
+profileAvatarFormValidator.enableValidation();
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
-const cardPopup = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+const cardPopup = new PopupWithForm(
+  "#add-card-modal",
+  // handleAddCardFormSubmit
+  (link, submitButton) => {
+    api.addCard(link, submitButton).then((res) => {
+      cardSection.addItem(res);
+    });
+    addFormValidator.toggleButtonState();
+    cardPopup.close();
+  }
+);
 cardPopup.setEventListeners();
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const profilePopup = new PopupWithForm(
+  "#profile-avatar-modal",
+  (url, submitButton) => {
+    // handles the avatar picture submit
+    api.setUserAvatar(url, submitButton).then((res) => {
+      profileImage.src = res.avatar;
+    });
+
+    profileAvatarFormValidator.toggleButtonState();
+    profilePopup.close();
+  }
+);
+profilePopup.setEventListeners();
+
 const userInfoPopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileFormSubmit
+  (values, submitButton) => {
+    userInfo.setUserInfo(values);
+    api.setUserInfo(values, submitButton);
+    userInfoPopup.close();
+  }
 );
-const cardSection = new Section(initialCards, renderCard, cardListEl);
-cardSection.renderItems();
-
 userInfoPopup.setEventListeners();
-//----------------functions
 
 function renderCard(cardData) {
-  const card = new Card(cardData, cardTemplate, handleImageClick);
-  return card.getView();
-}
+  const card = new Card(
+    cardData,
+    cardTemplate,
+
+    //handles the image click
+    (name, link) => {
+      imagePopup.open(name, link);
+    },
+    //handles the delete button click
+    (id, cardElement) => {
+      cardDeleteModal.open();
+      cardDeleteModal.setEventListeners();
```

Comment:
`.setEventListeners()` should be removed from `openings` because it's a **public** method that should be called only **once** in `index.js` otherwise you add listeners again and again to the same elements. This can cause a **`memory leak`**